tests/ui/fn/fn-item-type.stderr

```diff
@@ -2,8 +2,9 @@ error[E0308]: mismatched types
   --> $DIR/fn-item-type.rs:22:19
    |
 LL |     eq(foo::<u8>, bar::<u8>);
-   |     --            ^^^^^^^^^ expected fn item, found a different fn item
-   |     |
+   |     -- ---------  ^^^^^^^^^ expected fn item, found a different fn item
+   |     |  |
+   |     |  expected argument `y` to be a fn item because that argument needs to match the type of this parameter
```

Comment:
I feel like this is an improvement however, though maybe can be improved to mentioned the different fn item here too.